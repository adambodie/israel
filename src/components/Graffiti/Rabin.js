import React, { Component } from 'react'
import { Column } from 'react-foundation'

export default class Rabin extends Component {
	constructor() {
		super()
		this.state = {
			className: '',
			testId: 'hoverOff',
		}
		this.hoverOn = this.hoverOn.bind(this)
		this.hoverOff = this.hoverOff.bind(this)
	}
	hoverOn() {
		this.setState(prevState =>({ 
			className: 'hoverOver',
			testId: 'hoverOver' 
		}))
	}
	hoverOff() {
		this.setState(prevState =>({ 
			className: '',
			testId: 'hoverOff' 
		}))
	}
	render() {
		const { className, testId } = this.state
		return (
			<Column>
				<h1>The Most Important Piece of Graffiti in Tel Aviv</h1>
				<div className='rabin'>
					<img id='rabin1' src={`/assets/main/rabin-photo.jpg`} alt='Rabin Photograph' />
					<img id='rabin2' src={`/assets/main/rabin-painting.jpg`} alt='Rabin Painting' className={className} data-testid={testId} onMouseOver={()=>this.hoverOn()} onMouseLeave={()=>this.hoverOff()} />
				</div>
				<p>This piece of graffiti, as you hover over the image above, shows it is a painting of an actual photograph.  That photograph depicts the exact moment when Prime Minister Yitzhak Rabin was assassinated in 1995.  When someone attempted to paint over this mural, people got upset and started protesting.  Since then, it has been declared protected.</p>
			</Column>
		)
	}
}
