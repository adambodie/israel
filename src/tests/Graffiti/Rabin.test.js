import React from 'react';
import renderer from 'react-test-renderer';
import { StateMock } from '@react-mock/state';
import {
	render,
    cleanup,
    fireEvent,
    waitForElement
  } from 'react-testing-library'

import Rabin from '../../components/Graffiti/Rabin';

afterEach(cleanup) 

it('renders correctly', () => {
	const tree = renderer.create(<Rabin />).toJSON();
	expect(tree).toMatchSnapshot();
});

const renderComponent = ({ className, testId }) =>
  render(
    <StateMock state={{ className, testId }}>
      <Rabin/>
    </StateMock>
  );

  it('renders initial state', async () => {
    const { getByTestId } = renderComponent({ className: '', testId: 'hoverOff' });
    await waitForElement(() => getByTestId(/hoverOff/i));
  });

  it('update state', async () => {
    const { getByTestId } = renderComponent({ className: '', testId: 'hoverOff' });  
    fireEvent.mouseOver(getByTestId(/hoverOff/i));
    await waitForElement(() => getByTestId(/hoverOver/i));
    fireEvent.mouseLeave(getByTestId(/hoverOver/i));
    await waitForElement(() => getByTestId(/hoverOff/i));
  });