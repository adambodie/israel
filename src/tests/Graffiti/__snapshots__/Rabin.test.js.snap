// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="columns"
>
  <h1>
    The Most Important Piece of Graffiti in Tel Aviv
  </h1>
  <div
    className="rabin"
  >
    <img
      alt="Rabin Photograph"
      id="rabin1"
      src="/assets/main/rabin-photo.jpg"
    />
    <img
      alt="Rabin Painting"
      className=""
      data-testid="hoverOff"
      id="rabin2"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      src="/assets/main/rabin-painting.jpg"
    />
  </div>
  <p>
    This piece of graffiti, as you hover over the image above, shows it is a painting of an actual photograph.  That photograph depicts the exact moment when Prime Minister Yitzhak Rabin was assassinated in 1995.  When someone attempted to paint over this mural, people got upset and started protesting.  Since then, it has been declared protected.
  </p>
</div>
`;
